### PR TITLE
Add porting guide for Hardware CRC HAL module

### PR DESCRIPTION
**Pull request by scartmell-arm**

### CODE REVIEW

**Comment by AnotherButler** [comment]
@scartmell-arm Fantastic work on this. It looks really good 👍 

Could you please link the PR we're waiting on for the testing and class reference links?

**Comment by scartmell-arm** [comment]
https://github.com/ARMmbed/mbed-os/pull/6708

File: docs/reference/contributing/target/crc.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+## Hardware CRC
+
+The hardware CRC HAL API provides a low-level interface to the hardware CRC module of a target platform. Implementing the hardware CRC API allows you to gain the performance benefits of using hardware acceleration for CRC calculations. For platforms without hardware CRC capabilities, the API falls back to using the table and bitwise CRC implementations.
+
```
Comment (deepikabhavnani) [inline docs/reference/contributing/target/crc.md]: Note here is required for Hardware CRC - not thread safe

**Comment by AnotherButler** [comment]
@deepikabhavnani I've added that note. Does it look OK?

**Comment by AnotherButler** [comment]
@bulislaw Have your concerns been addressed? If so, is this OK to merge once the code merges?

**Comment by deepikabhavnani** [comment]
@AnotherButler - Should it be "The hardware CRC API's are not thread safe" ?

**Comment by bulislaw** [comment]
Not yet, but it passed Cis so should be merged very soon.

File: docs/reference/contributing/target/crc.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+The Mbed OS HAL API provides a set of conformance tests for Hardware CRC. You can use these tests to validate the correctness of your implementation. To run the hardware CRC HAL tests, use the following command:
+
+```
+mbed test -t <toolchain> -m <target> -n "tests-mbed_hal-crc*"
+```
+
+You can read more about the test cases here:
+
+```
+Fix link after code is merged.
```
Comment (AnotherButler) [inline docs/reference/contributing/target/crc.md]: Query: Where do we find the tests? I think they should be on this page? https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/modules.html

**Comment by MarceloSalazar** [comment]
Can we please finalize and merge this for OOB?

File: docs/reference/contributing/target/crc.md (hunk 2 above)
Comment (scartmell-arm) [reply docs/reference/contributing/target/crc.md]: The test still haven't been merged. The PR is here: https://github.com/ARMmbed/mbed-os/pull/6831

**Pull request merged**